ingest/src/main/scala/hydra.ingest/services/IngestionFlowV2.scala

```diff
@@ -108,26 +109,70 @@ object IngestionFlowV2 {
+  }
+
+  final case class RequiredGenericGenericRecordUnexpectedValueError(fields: List[(String, String)])
+    extends GenericRecordValueError {
+    override val message: String = s"""Required filed(s) with unexpected value(s) ${fields.map(kv => kv._1 + ":" + kv._2).mkString("[", ",", "]")}"""
+  }
+
   private[services] def validateKeyAndValueSchemas(key: GenericRecord, valueOpt: Option[GenericRecord]): Either[Throwable, Unit] = {
     valueOpt match {
       case Some(value) =>
         val keyAndValueMismatch = key.getSchema.getFields.asScala.toList.exists { k =>
           value.hasField(k.name()) && key.get(k.name()) != value.get(k.name())
         }
-        if (keyAndValueMismatch) {
-          val keyAndValueMismatch = key.getSchema.getFields.asScala.toList.flatMap { k =>
-            if (value.hasField(k.name()) && key.get(k.name()) != value.get(k.name())) {
-              Some(KeyAndValueMismatch(k.name(), key.get(k.name()), value.get(k.name())))
-            } else {
-              None
+        val requiredDateFields = List(RequiredField.CREATED_AT, RequiredField.UPDATED_AT)
+
+        val missedRequiredDateFields = requiredDateFields.diff(requiredDateFields.filter(requiredField => value.hasField(requiredField)))
+
+        val unexpectedValues = validateDateValues(value, requiredDateFields)
+
+        if (missedRequiredDateFields.nonEmpty) {
+          Left(RequiredGenericGenericRecordValueFieldMissingError(missedRequiredDateFields, value.toString))
+        } else if (unexpectedValues.nonEmpty) {
+          Left(RequiredGenericGenericRecordUnexpectedValueError(unexpectedValues))
+        } else {
+          if (keyAndValueMismatch) {
+            val keyAndValueMismatch = key.getSchema.getFields.asScala.toList.flatMap { k =>
+              if (value.hasField(k.name()) && key.get(k.name()) != value.get(k.name())) {
+                Some(KeyAndValueMismatch(k.name(), key.get(k.name()), value.get(k.name())))
+              } else {
+                None
+              }
             }
+            Left(KeyAndValueMismatchedValuesException(keyAndValueMismatch))
+          } else {
+            Right(())
           }
-          Left(KeyAndValueMismatchedValuesException(keyAndValueMismatch))
-        } else {
-          Right(())
         }
       case None =>
         Right(())
     }
   }
+  private def validateDateValues(value: GenericRecord, requiredDateFields: List[String]) = {
+    requiredDateFields.flatMap {
+      requiredField =>
+        value.getSchema.getFields.asScala.find(_.name == requiredField) match {
+          case Some(fieldValue) =>
+
+            val timestampMilliType = LogicalTypes.timestampMillis().addToSchema(Schema.create(Schema.Type.LONG))
+
+            val isLogicalType = fieldValue.schema().getType.equals(Schema.Type.LONG) &&
+              fieldValue.schema().getLogicalType.equals(timestampMilliType.getLogicalType)
+
+            if (isLogicalType) {
```

Comment:
Can be added as guard:
`value.getSchema.getFields.asScala.find(_.name == requiredField) match {
          case Some(fieldValue) if  isLogicalType=> ...  }`